### PR TITLE
Enable BlockVolume feature gate

### PR DESCRIPTION
**Pull request by slintes**
The primary goal of this PR is to enable the BlockVolume feature gate

For getting there I needed to update how we configure kubelet extra arguments. This ended in a bigger change for openshift: I converted the inventory from ini style to yaml. Main reason is that kubelet is configured with node groups now, and that would have been a very long json line in the ini config.

### CODE REVIEW

**Comment by slintes** [comment]
/cc @rmohr @cynepco3hahue 
can please have a look? :)

File: k8s-multus/scripts/node01.sh [hunk 1]
```diff
@@ -16,4 +15,22 @@ kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /etc/kubernetes/flanne
...
+    sleep 10
+    echo "Waiting for api server to be available..."
+    kubectl --kubeconfig=/etc/kubernetes/admin.conf get nodes --no-headers
+    kubectl_rc=$?
+    retry_counter=$((retry_counter + 1))
+done
+kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/local-volume.yaml
+
+# create local block device, backed by raw cirros disk image (see also provision.sh)
+LOOP_DEVICE=`losetup --find --show /mnt/local-storage/cirros.img.raw`
```
Comment (cynepco3hahue) [inline k8s-multus/scripts/node01.sh]: Can it be part of https://github.com/kubevirt/kubevirt/tree/master/images/disks-images-provider?

File: k8s/scripts/provision.sh [hunk 2]
```diff
@@ -142,3 +158,7 @@ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
 # Pre pull fluentd image used in logging
 docker pull fluent/fluentd:v1.2-debian
 docker pull fluent/fluentd-kubernetes-daemonset:v1.2-debian-syslog
+
+# convert cirros image and copy it into the vm (used in node01.sh)
+qemu-img convert -f qcow2 -O raw /tmp/cirros.img /tmp/cirros.img.raw
```
Comment (cynepco3hahue) [inline k8s/scripts/provision.sh]: the same

File: os-3.10-multus/scripts/node01.sh [hunk 3]
```diff
@@ -69,15 +57,39 @@ cat >post_deployment_configuration <<EOF
...
-
 /usr/bin/oc create -f /tmp/macvlan-conf.yaml
-
 /usr/bin/oc create -f /tmp/ptp-conf.yaml
+/usr/bin/oc create -f /tmp/bridge-conf.yaml
 
-/usr/bin/oc create -f /tmp/bridge-conf.yaml
\ No newline at end of file
+# create local block device, backed by raw cirros disk image (see also provision.sh)
+LOOP_DEVICE=`losetup --find --show /mnt/local-storage/cirros.img.raw`
```
Comment (cynepco3hahue) [inline os-3.10-multus/scripts/node01.sh]: the same

File: os-3.10-multus/scripts/provision.sh [hunk 4]
```diff
@@ -153,3 +221,7 @@ docker pull docker.io/nfvpe/multus
 docker pull quay.io/schseba/l2-bridge-cni-plugin
 docker pull quay.io/schseba/cni-plugins
 docker pull quay.io/external_storage/local-volume-provisioner:v2.1.0
+
+# convert cirros image and copy it into the vm (used in node01.sh)
+qemu-img convert -f qcow2 -O raw /tmp/cirros.img /tmp/cirros.img.raw
```
Comment (cynepco3hahue) [inline os-3.10-multus/scripts/provision.sh]: the same

File: os-3.10/scripts/provision.sh [hunk 5]
```diff
@@ -141,3 +209,8 @@ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
 # Pre pull fluentd image used in logging
 docker pull docker.io/fluent/fluentd:v1.2-debian
 docker pull fluent/fluentd-kubernetes-daemonset:v1.2-debian-syslog
+
+# convert cirros image and copy it into the vm (used in node01.sh)
+qemu-img convert -f qcow2 -O raw /tmp/cirros.img /tmp/cirros.img.raw
```
Comment (cynepco3hahue) [inline os-3.10/scripts/provision.sh]: the same

File: k8s-multus/scripts/node01.sh (hunk 1 above)
Comment (slintes) [reply k8s-multus/scripts/node01.sh]: sounds like a good idea, thx :)

Comment (slintes) [reply k8s-multus/scripts/node01.sh]: got it working with disks-images-provider, will remove this stuff from CI in a minute

**Comment by slintes** [comment]
I removed the setup of the cirros block device and will add it to kubevirt's disks-images-provider (already tested it successfully locally)

**Comment by slintes** [comment]
let me know if the other parts are ok for you, I will build and push all images then, and test them in another PR in the kubevirt repository. Didn't run all tests locally yet, because that will take ages ;)

**Comment by slintes** [comment]
I just added a provisionAndPublishAll.sh script, example output:

```
$ ./provisionAndPublishAll.sh

Provisioning and publishing these images, this will take a while:
k8s/1.10.4 k8s/1.11.0 k8s-multus/1.10.4 k8s-multus/1.11.1 os-3.10 os-3.10-crio os-3.10-multus

All logs will go to [...]/kubevirt.io/kubevirtci/log.txt, here you will only see the final shas for each image!

Do you want to continue? [y/n] y

Thu Sep 13 12:25:51 CEST 2018
building k8s/1.10.4
The push refers to repository [docker.io/kubevirtci/k8s-1.10.4]
latest: digest: sha256:9341db6774ccc7d81a2fe90417fcced02867d7109e7c21a12dc6791b2746cccc size: 2003

Thu Sep 13 12:32:56 CEST 2018
building k8s/1.11.0
...
```

**Comment by slintes** [comment]
Hey, I updated the shas, and they are already used in https://github.com/kubevirt/kubevirt/pull/1511.
To me the test failures there seem to be unrelated. Can we merge this and the PR above? I need it for pushing the PR which implements the Kubevirt block device feature.

/cc @rmohr

**Pull request merged**